source/core/Native.cs

```diff
@@ -0,0 +1,711 @@
+            {
+                return new InputArgument(new string(value));
+            }
+
+            // Pointer types
+            public static implicit operator InputArgument(IntPtr value)
+            {
+                return new InputArgument(value);
+            }
+            public static unsafe implicit operator InputArgument(void* value)
+            {
+                return new InputArgument(new IntPtr(value));
+            }
+
+            public static implicit operator InputArgument(OutputArgument value)
+            {
+                return new InputArgument(value._storage);
+            }
+        }
+
+        public class OutputArgument
+        {
+            internal IntPtr _storage = Marshal.AllocCoTaskMem(24);
+            private bool disposed;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="OutputArgument"/> class for natives that output data into pointers.
+            /// </summary>
+            public OutputArgument()
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="OutputArgument"/> class with an initial value for natives that require the pointer to data instead of the actual data.
+            /// </summary>
+            /// <param name="initvalue">The value to set the data of this <see cref="OutputArgument"/> to.</param>
+            public OutputArgument(object value) : this()
+            {
+                unsafe
+                {
+                    *(ulong*)(_storage) = Function.ObjectToNative(value);
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!disposed)
+                {
+                    if (disposing)
+                    {
+                    }
+                    Marshal.FreeCoTaskMem(_storage);
+                    disposed = true;
+                }
+            }
+            ~OutputArgument()
+            {
+                Dispose(false);
+            }
+
+            /// <summary>
+            /// Gets the value of data stored in this <see cref="OutputArgument"/>.
+            /// </summary>
+            public T GetResult<T>()
+            {
+                unsafe
+                {
+                    return (T)(Function.ObjectFromNative(typeof(T), (ulong*)(_storage.ToPointer())));
+                }
+            }
+        }
+
+        public sealed class Function
+        {
+            internal unsafe struct NativeTask : IScriptTask
+            {
+                internal ulong _hash;
+                internal ulong* _result;
+                internal InputArgument[] _arguments;
+
+                public void Run()
+                {
+					Wrapper.NativeInit(_hash);
+
+                    foreach (var argument in _arguments)
```

Comment:
Space vs Tabs 😄 . Note that the "Productivity Power Tools" extension for Visual Studio (which every developer should have installed) asks to fix mixed spaces/tabs when you open a file, so it's really easy to get those sorted out.

Reply:
I didn't know "Productivity Power Tools" had that feature, thanks man.

Reply:
I didn't know "Productivity Power Tools" had that feature, thanks man.